fiftyone/utils/map.py

```diff
@@ -13,13 +13,15 @@
 
 def map_samples(
     sample_collection,
-    map_fcn,
-    workers=None,
-    batch_method="id",
-    progress=None,
-    save=False,
-    parallelize_method="process",
-    skip_failures=True,
+    map_fcn: Callable[[Any], None],
+    workers: Optional[int] = None,
+    batch_method: Union[Literal["id"], Literal["slice"]] = "id",
+    progress: Optional[Union[bool, Literal["worker"]]] = None,
+    save: bool = False,
+    parallelize_method: Union[
+        Literal["process"], Literal["thread"]
+    ] = "process",
+    skip_failures: bool = False,
```

Comment:
I took the types out here to be more in line with the rest of `fiftyone` for now. But if we're gonna keep them let's ensure they are correct and as simplified as possible.

```suggestion
    map_fcn: Callable[[Any], Any],
    workers: Optional[int] = None,
    batch_method: str = "id",    
    progress: Optional[Union[bool, Literal["worker"]]] = None,
    save: bool = False,
    parallelize_method: str = "process",
    skip_failures: bool = False,
```

`batch_method` and `parallelize_method` are both strings because they use a factories (so we can easily add more without updating this file).

Reply:
I see OK. I don't think we will be updating `batch_methods` and `parallelize_method` very often, so it might be OK to use literal with finite set; it makes better type hinting in my opinion :)